core/save-html.cpp

```diff
@@ -92,7 +92,8 @@ static void write_dive_status(struct membuffer *b, const struct dive &dive)
 
 static void put_HTML_bookmarks(struct membuffer *b, const struct dive &dive)
 {
-	const char *separator = "\"events\":[";
```

Comment:
Actually I think this is the wrong fix.

The bug was introduced in 27dbdd35c6d5d9d8c2d3c607405376ef4db6e60f when removing the check for "no events".

I would just add
```
        if (dive.dcs[0].events.empty())
                return;
```
to restore the old behavior.

Adding the new version on top might be a good idea anyway - the old code was very confusing.

Reply:
That said - I just quickly skimmed over the source file and it is very inconsistent on how to treat empty lists. :( This should be unified.

Reply:
I'd agree that this isn't the only possible fix - I'm just not sure if it's the _wrong_ fix.
I see no strong argument either way... do we want an empty array for the events (ensuring that the events array always exists), or is it better not to have the array. I see other spots in the code where we are making the wrong choice in this matter. For example, the html dive list table doesn't render correctly because elements are missing instead of being empty.

Reply:
Yes, with "wrong" I meant not restoring old behavior before the breaking commit.

As I said, this should all be unified.  I would go as far as writing a template that outputs a list. Something like `void output_list(const char *name, const T &list, FUNC output_function);`
Optionally a `bool output_empty_list = false` parameter.